### ISSUE TITLE
Multiple Reacts loading

### ISSUE THREAD

**Issue by genyded** [thereactivestack/kickstart-hugeapp#37]
Somehow when you build this now, React is installed in mode_modules even though it's not in the package file...

![image](https://cloud.githubusercontent.com/assets/3081676/11886219/8991a5fc-a4ed-11e5-8856-ab3cd786a351.png)

...and since Meteor loads it too we end with conflicts...

`Uncaught Error: Invariant Violation: addComponentAsRefTo(...): Only a ReactOwner can have refs. You might be adding a ref to a component that was not created inside a component's `render` method, or you have multiple copies of React loaded`

**Comment by genyded** [comment]
Not sure if this is related, but [this issue](https://github.com/meteor/react-packages/issues/135) was to seperate out react DOM in meteor and it seems based on [this comment](https://gist.github.com/jimfb/4faa6cbfb1ef476bd105#gistcomment-1605418) it may be related if react DOM is not being externalized?

**Comment by genyded** [comment]
The error only seems to come up when React DOM is involved

**Comment by eXon** [comment]
React is installed on the node_modules folder strictly for peer dependencies. It should not be used. In fact, if the react-runtime package is installed, it will detect the export and use it.

**Comment by eXon** [comment]
React is installed on the node_modules folder strictly for peer dependencies. It should not be used. In fact, if the react-runtime package is installed, it will detect the export and use it.

**Comment by genyded** [comment]
Well something is off - if I go back to the 'babel-5' version of the hugeapp and run the same code - there are no errors...

**Comment by eXon** [comment]
Do you have any code I can try to see your problem?

**Comment by genyded** [comment]
I am adding `"material-ui": "^0.14.0-rc2"` via npm  and you'll also need to add coniel:react-tap-event-plugin Meteor package. Then on any page/route open a Dialog...


````javascript
import { Component, PropTypes } from 'react';
import FlatButton from 'material-ui/lib/flat-button';
import Dialog from 'material-ui/lib/dialog';
import TextField from 'material-ui/lib/text-field';

injectTapEventPlugin();

export default class LoginDialog extends Component {

  constructor(props) {
    super(props);
  }

  static propTypes = {
    //cancelLogin: PropTypes.func.isRequired,
    //submitLogin: PropTypes.func.isRequired
  }

  render() {
      //Custom Actions
      let loginActions = [
        <FlatButton
          label="Cancel"
          secondary={true}
          onTouchTap={this.props.cancelLogin} />,
        <FlatButton
          label="Login"
          primary={true}
          onTouchTap={this.props.submitLogin}
        />
      ];

      return (
      <Dialog {...this.props}
        key="loginDialog"
        title="ePapers Login" actions={loginActions}
          open={true}>
        <TextField
          type="email"
          key="emailField"
          floatingLabelText="Email"
          type="email"
          fullWidth={true}
          autoFocus
        />
        <TextField
          key="passwordField"
          floatingLabelText="Password"
          type="password"
          fullWidth={true}
        />
      </Dialog>
    );
  }
}
````

With the latest hugeapp, you should see the error in the console and more if you click any buttons. With the babel-5 version of hugeapp it all works fine (none of the errors) except I disabled the button clicks in the code above so they won't actually work no matter what.

I expect you can see the same issue if you invoke anything that uses reacts life-cycle methods or react-DOM though.

**Comment by genyded** [comment]
Any progress on this?

**Comment by eXon** [comment]
I think this problem is linked with material-ui only. I'll take a look.

**Comment by genyded** [comment]
The odd things is that with the Babel 5 version of huheapp, the same code with the same Material UI version works without any errors.

**Comment by genyded** [comment]
OK, if I add this section into entry/webpack,conf,js...

```javascript
    externals: {
    // Make sure we use Meteor package for jQuery, react and react-router
    'jquery': '$',
    'react': 'React',
    'react-router': 'ReactRouter',
    'react-dom': 'ReactDOM',
    'react-addons-transition-group': 'React.addons.TransitionGroup',
    'react-addons-create-fragment': 'React.addons.createFragment',
    'react-addons-pure-render-mixin': 'React.addons.PureRenderMixin',
    'react-addons-update': 'React.addons.update',
    'react-addons-linked-state-mixin': 'React.addons.LinkedStateMixin',
    'react-addons-test-utils': 'React.addons.TestUtils'
  },````

,,, the errors go away and everything works. Is there a reason that section was taken out when moving from babel 5 to babel 6?

**Comment by genyded** [comment]
See my last comment, as long as there is not real reason not to add that back into the webpack config, this can be closed. If there is an issue with the above or you want to use this as a reminder to add that back into the source, please reopen.

**Comment by dbismut** [comment]
I got the same problem with material-ui and solved it the same way @genyded did.

**Comment by eXon** [comment]
This must be solved. So many people are using material-ui I'll get a gazilion issue about it.

**Comment by gitupiikki** [comment]
genyded: Thanks for your fix. I'm not even using material UI atm but this seems help. Being struggling with this few days :/

**Comment by eXon** [comment]
This is not about Babel 5 or 6. This is because we now generate React externals automatically. Those are little bit more tricky to fix but in meteor-webpack 1.0, they will be. For now, just keep them in your externals.

**Comment by minfawang** [comment]
It will be sooo awesome to see 1.0  :grinning:
